packages/cli/lib/developerTestAccountCreate.js

```diff
@@ -0,0 +1,194 @@
+  try {
+    const token = await getAccessToken(personalAccessKey, env);
+    updatedConfig = await updateConfigWithAccessToken(
+      token,
+      personalAccessKey,
+      env
+    );
+  } catch (e) {
+    logErrorInstance(e);
+  }
+  if (!updatedConfig) {
+    throw new Error('Failed to update config with personal access key.');
+  }
+
+  let validName = updatedConfig.name;
+  if (!updatedConfig.name) {
+    const nameForConfig = result.accountName
+      .toLowerCase()
+      .split(' ')
+      .join('-');
+    validName = nameForConfig;
+    const invalidAccountName = accountNameExistsInConfig(nameForConfig);
+    if (invalidAccountName) {
+      if (!force) {
+        logger.log('');
+        logger.warn(
+          i18n(
+            `cli.lib.prompts.enterAccountNamePrompt.errors.accountNameExists`,
+            { name: nameForConfig }
+          )
+        );
+        const { name: promptName } = await enterAccountNamePrompt(
+          nameForConfig + `_${result.id}`
+        );
+        validName = promptName;
+      } else {
+        // Basic invalid name handling when force flag is passed
+        validName = nameForConfig + `_${result.id}`;
+      }
+    }
+  }
+
+  updateAccountConfig({
+    ...updatedConfig,
+    environment: updatedConfig.env,
+    tokenInfo: updatedConfig.auth.tokenInfo,
+    name: validName,
+  });
+  writeConfig();
+
+  logger.log('');
+  return validName;
+};
+
+const buildDeveloperTestAccount = async ({
+  name,
+  accountConfig,
+  env,
+  maxTestPortals,
+  force = false,
+}) => {
+  SpinniesManager.init({
+    succeedColor: 'white',
+  });
+  const accountId = getAccountId(accountConfig.portalId);
+
+  let result;
+  const spinniesI18nKey = `${i18nKey}.create.loading`;
+
+  try {
+    logger.log('');
+    SpinniesManager.add('devTestAcctCreate', {
+      text: i18n(`${spinniesI18nKey}.add`, {
+        accountName: name,
+      }),
+    });
+    result = await createDeveloperTestAccount(accountId, name);
+
+    SpinniesManager.succeed('devTestAcctCreate', {
+      text: i18n(`${spinniesI18nKey}.succeed`, {
+        name: result.accountName,
+        accountId: result.id,
+      }),
+    });
+  } catch (err) {
+    debugErrorAndContext(err);
+
+    SpinniesManager.fail('devTestAcctCreate', {
+      text: i18n(`${spinniesI18nKey}.fail`, {
+        accountName: name,
+      }),
+    });
+
+    if (isMissingScopeError(err)) {
+      logger.error(
+        i18n(`${i18nKey}.create.failure.scopes.message`, {
+          accountName: uiAccountDescription(accountId),
+        })
+      );
+      const websiteOrigin = getHubSpotWebsiteOrigin(env);
+      const url = `${websiteOrigin}/personal-access-key/${accountId}`;
+      logger.info(
+        i18n(`${i18nKey}.create.failure.scopes.instructions`, {
+          accountName: uiAccountDescription(accountId),
+          url,
+        })
+      );
+    } else if (
+      isSpecifiedError(err, {
+        statusCode: 400,
+        errorType: 'TEST_PORTAL_LIMIT_REACHED',
```

Comment:
Dev test accounts BE returns a slightly different error format which doesnt have a category/subcategory, so I had to update the isSpecifiedError def in LDL to support this